ruby/ql/lib/codeql/ruby/controlflow/internal/Completion.qll

```diff
@@ -227,6 +227,8 @@ private predicate inMatchingContext(AstNode n) {
   or
   n instanceof CasePattern
   or
+  n = any(CasePattern c).getDesugared()
```

Comment:
@hvitved Is this something that should be addressed more generally in the CFG library? I guess we'd want to preserve the completion context when desugaring a node. Should `isValidFor(node) implies isValidFor(node.getDesugared())` hold in general?

Reply:
Yes, that sounds right.

Reply:
Does this look right to you? https://github.com/github/codeql/pull/7283/commits/06b3d583e512f11f37f9320517cef9d7f3dbc13f